src/actions/receive-profile.js

```diff
@@ -1409,16 +1407,14 @@ export function getProfilesFromRawUrl(
     }
     dispatch(setDataSource(dataSource));
 
-    let shouldSetupInitialUrlState = true;
     switch (dataSource) {
       case 'from-addon':
-        shouldSetupInitialUrlState = false;
         // We don't need to `await` the result because there's no url upgrading
         // when retrieving the profile from the addon and we don't need to wait
         // for the process. Moreover we don't want to wait for the end of
         // symbolication and rather want to show the UI as soon as we get
         // the profile data.
-        dispatch(retrieveProfileFromAddon());
+        dispatch(retrieveProfileFromAddon()).catch(() => {});
```

Comment:
I also added a `catch` here. The test I added started to complain about uncaught promise rejections. So added this to handle any errors that can be thrown. You can say that we shouldn't just ignore the errors. but actually we throw a fatal error here already inside the `retrieveProfileFromAddon` function:
https://github.com/firefox-devtools/profiler/blob/a52f10041dd4bdf5f972e394d50385c4b54fd897/src/actions/receive-profile.js#L925-L928
So we don't have to handle the error twice here.

Reply:
Thanks for the suggestion. Yeah, that solution looks a lot better. Changed the code.